frontend/lib/resolution/default-functions.cpp

```diff
@@ -320,13 +323,15 @@ generateDeinitSignature(Context* context, const CompositeType* inCompType) {
   const CompositeType* compType = nullptr;
   std::vector<UntypedFnSignature::FormalDetail> ufsFormals;
   std::vector<QualifiedType> formalTypes;
+  gdbShouldBreakHere();
 
-  generateInitParts(context, inCompType, compType, ufsFormals, formalTypes);
+  generateInitParts(context, inCompType, compType,
+                    ufsFormals, formalTypes, /*useGeneric*/ false);
 
   // build the untyped signature
   auto ufs = UntypedFnSignature::get(context,
                         /*id*/ compType->id(),
-                        /*name*/ USTR("init"),
+                        /*name*/ USTR("deinit"),
```

Comment:
I think we got away with it because ``considerCompilerGeneratedCandidates`` asks for a TypedFnSignature and didn't expect to get something with the wrong name. I just pushed a change that also fixes the name of ``init=`` and adds an assertion to ``getCompilerGeneratedMethodQuery``.